services/crypto_approve_allowance.proto

```diff
@@ -26,28 +26,114 @@ option java_package = "com.hederahashgraph.api.proto.java";
 option java_multiple_files = true;
 
 import "basic_types.proto";
+import "google/protobuf/wrappers.proto";
 
 /**
- * Creates one or more hbar/token approved allowances <b>relative to the payer account of this
- * transaction</b>. Each allowance grants a spender the right to transfer a pre-determined 
- * amount of the payer's hbar/token to any other account of the spender's choice. 
+ * Creates one or more hbar/token approved allowances <b>relative to the owner account specified in the allowances of
+ * this transaction</b>. Each allowance grants a spender the right to transfer a pre-determined amount of the owner's
+ * hbar/token to any other account of the spender's choice. If the owner is not specified in any allowance, the payer
+ * of transaction is considered to be the owner for that particular allowance.
+ * Setting the amount to zero in CryptoAllowance or TokenAllowance will remove the respective allowance for the spender.
  * 
- * (So if account <tt>0.0.X</tt> pays for this transaction, then at consensus each spender 
- * account will have new allowances to spend hbar or tokens from <tt>0.0.X</tt>).
+ * (So if account <tt>0.0.X</tt> pays for this transaction and owner is not specified in the allowance,
+ * then at consensus each spender account will have new allowances to spend hbar or tokens from <tt>0.0.X</tt>).
  */
 message CryptoApproveAllowanceTransactionBody {
-    /**
-     * List of hbar allowances approved by the account owner.
-     */
-    repeated CryptoAllowance cryptoAllowances = 1;
-
-    /**
-     * List of non-fungible token allowances approved by the account owner.
-     */
-    repeated NftAllowance nftAllowances = 2;
-
-    /**
-     * List of fungible token allowances approved by the account owner.
-     */
-    repeated TokenAllowance tokenAllowances = 3;
+  /**
+   * List of hbar allowances approved by the account owner.
+   */
+  repeated CryptoAllowance cryptoAllowances = 1;
+
+  /**
+   * List of non-fungible token allowances approved by the account owner.
+   */
+  repeated NftAllowance nftAllowances = 2;
+
+  /**
+   * List of fungible token allowances approved by the account owner.
+   */
+  repeated TokenAllowance tokenAllowances = 3;
 }
+
+/**
+ * An approved allowance of hbar transfers for a spender.
+ */
+message CryptoAllowance {
```

Comment:
Do these need to be defined before `CryptoApproveAllowanceTransactionBody` to compile?

Reply:
It seems to compile fine this way as well